content/en/docs/tasks/configure-pod-container/resize-container-resources.md

```diff
@@ -2,264 +2,192 @@
     - resourceName: cpu
       restartPolicy: NotRequired
     - resourceName: memory
       restartPolicy: RestartContainer
-    resources:
-      limits:
-        memory: "200Mi"
-        cpu: "700m"
-      requests:
-        memory: "200Mi"
-        cpu: "700m"
 ```
 
+* `NotRequired`: (Default) Apply the resource change to the running container without restarting it.
+* `RestartContainer`: Restart the container to apply the new resource values. This is often necessary for memory changes because many applications and runtimes cannot adjust their memory allocation dynamically.
+
+If `resizePolicy[*].restartPolicy` is not specified for a resource, it defaults to `NotRequired`.
+
 {{< note >}}
-In the above example, if desired requests or limits for both CPU _and_ memory
-have changed, the container will be restarted in order to resize its memory.
+If a Pod's overall `restartPolicy` is `Never`, then any container `resizePolicy` must be `NotRequired` for all resources. You cannot configure a resize policy that would require a restart in such Pods.
 {{< /note >}}
 
-<!-- steps -->
+**Example Scenario:**
+
+Consider a container configured with `restartPolicy: NotRequired` for CPU and `restartPolicy: RestartContainer` for memory.
+* If only CPU resources are changed, the container is resized in-place.
+* If only memory resources are changed, the container is restarted.
+* If *both* CPU and memory resources are changed simultaneously, the container is restarted (due to the memory policy).
 
 ## Limitations
 
-In-place resize of pod resources currently has the following limitations:
+In-place resource resizing currently has the following limitations:
+
+* **Resource Types:** Only CPU and memory resources can be resized.
+* **Memory Decrease:** Memory limits _cannot be decreased_ unless the `resizePolicy` for memory is `RestartContainer`. Memory requests can generally be decreased.
+* **QoS Class:** The Pod's original [Quality of Service (QoS) class](/docs/tasks/configure-pod-container/quality-service-pod/) (Guaranteed, Burstable, BestEffort) is determined at creation and cannot be changed by a resize. The resized resource values must still adhere to the rules of the original QoS class:
+    * *Guaranteed*: Requests must continue to equal limits for both CPU and memory after resizing.
+    * *Burstable*: Requests and limits cannot become equal for *both* CPU and memory simultaneously (as this would change it to Guaranteed).
+    * *BestEffort*: Resource requirements (`requests` or `limits`) cannot be added (as this would change it to Burstable or Guaranteed).
+* **Container Types:** Non-restartable Init Containers and Ephemeral Containers cannot be resized. [Sidecar containers](/docs/concepts/workloads/pods/sidecar-containers/) can be resized.
+* **Resource Removal:** Resource requests and limits cannot be entirely removed once set; they can only be changed to different values.
+* **Operating System:** Windows pods do not support in-place resize.
```

Comment:
Is this true:
```suggestion
* **Operating System:** Pods with the OS set to Windows do not support in-place resize. If you attempt to resize a Pod that is bound
  to a Windows node, you also cannot resize it. However, you can attempt to resize a Pod that uses a Windows container, provided it has been created but is not yet bound to a node.
```
?

Reply:
We attempt to enforce this in admission, so the apiserver will reject the request if the OS is set to Windows. In the second case, the request will go through, but the resize is marked as infeasible, so it's still not supported. I didn't distinguish in this list which limitations are enforced in admission and which are enforced by the Kubelet. That level of detail seems unnecessary?

Reply:
Is this better?
```suggestion
* **Operating System:** Pods on Windows nodes cannot be resized.
```

Reply:
Oh, I missed this part

> However, you can attempt to resize a Pod that uses a Windows container, provided it has been created but is not yet bound to a node.

That is technically true, but IMO we shouldn't document it. It's an edge case that we don't want to encourage.